packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts

```diff
@@ -241,6 +241,27 @@ export interface AlbControllerOptions {
    * @default - Corresponds to the predefined version.
    */
   readonly policy?: any;
+
+  /**
+   * The values to be used by the chart.
+   * For nested values use a nested dictionary. For example:
+   * values: {
+   *  autoscaling: false,
+   *  ingressClassParams: { create: true }
+   * }
+   *
+   * Note that the following values are set by the controller and cannot be overridden:
+   *  - clusterName
+   *  - serviceAccount.create
+   *  - serviceAccount.name
+   *  - region
+   *  - vpcId
+   *  - image.repository
+   *  - image.tag
```

Comment:
Do we know the full set of values that can be passed in to the `helmChart.values` property? If so could we add some stronger typing checks to them? If not then could we add some validation for the values that cannot be overridden? 

From your unit test case it seems like Helm Chart will just ignore these values but I would rather prevent these values from being set at all instead of allowing it to be passed in and have it silently fail.

Reply:
i think the set of values varies (or can potentially vary) depending on the helm chart version. I think defining a type (e.g. AlbControllerHelmChartValues) will create a loose link to the helm chart that has the potential for breaking. Validating the passed values against the keys we know we explicitly set seems like a better approach to me.

Reply:
Gotcha, thanks for clarifying. Then let's go with the second option to validate the passed values that cannot be overridden.

Reply:
@paulhcsun i've updated with the requested changes. Assuming the suite passes, would you (or someone else) be able to run/update the integration tests? I ran into some issues with the teardown last time.

Reply:
Hey @mtrspringer, thanks for making the changes! and ya for sure, I'll give that a run later today.

Reply:
Hey @mtrspringer, apologies some other items came up. I will run the tests now.

Reply:
It took a while to run on Friday but looks like it failed due to a credentials related error. Possibly because it hung and timed out? I will try running it again but if it still fails then I'll ask another team member to help look into this as I'll be oncall this week.

Reply:
@paulhcsun thanks very much! yes it takes ages, looks like the build is passing though